src/modules/chat/index.js

```diff
@@ -341,11 +344,35 @@ class ChatModule {
               (!detectedEmote && partMetadataItem.type === 'prefix') ||
               (detectedEmote && partMetadataItem.type === 'suffix')
             ) {
+              predecessor = k;
               break;
             }
             modifiers.push(partMetadataItem);
             parts[k] = null;
           }
+
+          const overlayModifierPredicate = ({modifier}) => modifier === 'o!';
+          const negate = (pred) => (x) => !pred(x);
+
+          // An emote may be an overlay if it satisfies the following conditions:
+          // 1. It has a "predecessor" part.
+          // 2. That predecessor part is an emote.
+          // 3. That predecessor emote is not an overlay.
+          // 4. The user has applied the overlay modifier.
+          // This ensures that overlays do not stack and they only stack on another emote.
+          if (
+            predecessor != null &&
+            partMetadata[predecessor]?.emote &&
+            !isOverlay[predecessor] &&
+            modifiers.some(overlayModifierPredicate)
+          ) {
+            // Keep track of this part being an overlay.
+            isOverlay[j] = true;
+          } else {
+            // Strip the overlay modifier from the part.
+            modifiers = modifiers.filter(negate(overlayModifierPredicate));
```

Comment:
stripping it works, but it also results in the modifier going missing from the output. for other modifiers, we do not show the modified emote but we show the modifier was sent when used incorrectly. is there a way to replicate this behavior here?

![image](https://github.com/night/betterttv/assets/2942184/69cdbae2-66d4-47d4-8745-aabd60ae6172)

Reply:
as an aside, do you have an image asset for this modifier? if not, we can probably have one made

Reply:
> is there a way to replicate this behavior here?

I've modified it so that if an overlay modifier is removed, it replaces the corresponding parts in the part array.
![Screenshot 2024-02-17 094835](https://github.com/night/betterttv/assets/2078559/fa20500a-c871-4174-849c-3b88b686344e)

> do you have an image asset for this modifier?

I'm no artist, so feel free to use or not use this.
![overlay2](https://github.com/night/betterttv/assets/2078559/71da1601-14d7-4888-9457-108eb13b059d)

Reply:
@dclstn any ideas on the icon for this?

Reply:
Something like so perhaps:

<img src="https://github.com/night/betterttv/assets/43322006/78437ad9-0355-4494-9853-2ff34b649e52" width="250px" height="250px"/>

Reply:
When the image is small in emote form, it kind of just looks like 1 square. Maybe make it so that the top square doesn't overlap the bottom one quite so completely?

Reply:
agreed with Dallas' feedback. you can't tell what this is when smaller. it also resembles a copy button, which is not be the intention.

Reply:
How's this?
<img src="https://github.com/night/betterttv/assets/43322006/795a1cbf-c245-4540-aaba-5235b1e11bf9" width="256px"/>

downsized for reference:
<img src="https://github.com/night/betterttv/assets/43322006/795a1cbf-c245-4540-aaba-5235b1e11bf9" width="112px"/>
<img src="https://github.com/night/betterttv/assets/43322006/795a1cbf-c245-4540-aaba-5235b1e11bf9" width="56px"/>
<img src="https://github.com/night/betterttv/assets/43322006/795a1cbf-c245-4540-aaba-5235b1e11bf9" width="28px"/>

Reply:
this looks better